### PR TITLE
Allow to create Headers from Map

### PR DESCRIPTION
**Pull request by joschi**
This change set adds a method to the `Headers` class which allows users to create a new `Headers` instance from a `Map<String, String>` (similar to `Headers#of(String[])`).

### CODE REVIEW

File: okhttp-tests/src/test/java/com/squareup/okhttp/internal/http/HeadersTest.java [hunk 1]
```diff
@@ -234,4 +238,68 @@
     } catch (IllegalArgumentException expected) {
     }
   }
+
+  @Test public void ofMapThrowsOnNull() {
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/com/squareup/okhttp/internal/http/HeadersTest.java]: Nice tests!

File: okhttp/src/main/java/com/squareup/okhttp/Headers.java [hunk 2]
```diff
@@ -167,6 +168,33 @@ public static Headers of(String... namesAndValues) {
...
+  public static Headers of(Map<String, String> headers) {
+    if (headers == null) {
+      throw new IllegalArgumentException("Expected map with header names and values");
+    }
+
+    // Make a defensive copy and clean it up.
+    final String[] namesAndValues = new String[headers.size() * 2];
+    int i = 0;
+    for (Map.Entry<String, String> header : headers.entrySet()) {
+      if (header.getKey() == null || header.getValue() == null)
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/Headers.java]: nitpick: our code style always uses `{` for wrapped lines, and never uses `final` for locals

**Comment by swankjesse** [comment]
Nice.

Please accept our contributor's agreement, address my nitpicks & I'll merge your change.
https://docs.google.com/forms/d/13WR8m5uZ2nAkJH41k7GdVBXAAbzDk00vxtEYjd6Imzg/viewform?ndplr=1&formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ

**Comment by joschi** [comment]
@swankjesse I've signed the CLA and adapted to the project's code style.

**Comment by swankjesse** [comment]
Thanks!

**Pull request merged**